lib/modules/manager/npm/extract/locked-versions.ts

```diff
@@ -83,7 +83,10 @@ export async function getLockedVersions(
       for (const dep of packageFile.deps) {
         // TODO: types (#7154)
         dep.lockedVersion = semver.valid(
-          lockFileCache[npmLock].lockedVersions[dep.depName!]
+          lockFileCache[npmLock].lockedVersions[dep.depName!] ||
```

Comment:
```suggestion
          lockFileCache[npmLock].lockedVersions[dep.depName!] ??
```
can we instead check the lockfile version and make it conditional, for easier extension later.